src/internal/m365/collection/drive/helper_test.go

```diff
@@ -996,6 +1022,78 @@ func (dd *deltaDrive) newEnumer() *DeltaDriveEnumerator {
 	return &DeltaDriveEnumerator{Drive: clone}
 }
 
+type drivePrevPaths struct {
+	id                 string
+	folderIDToPrevPath map[string]string
+}
+
+func (dd *deltaDrive) newPrevPaths(
+	idPathPairs ...string,
+) *drivePrevPaths {
+	dpp := drivePrevPaths{
+		id:                 dd.id,
+		folderIDToPrevPath: map[string]string{},
+	}
+
+	if len(idPathPairs)%2 == 1 {
+		dpp.folderIDToPrevPath["error"] = "idPathPairs had odd count of elements"
+		return &dpp
+	}
+
+	for i := 0; i < len(idPathPairs); i += 2 {
+		dpp.folderIDToPrevPath[idPathPairs[i]] = idPathPairs[i+1]
+	}
+
+	return &dpp
+}
+
+// transforms 0 or more drivePrevPaths to a map[driveID]map[folderID]prevPathString
+func multiDrivePrevPaths(drivePrevs ...*drivePrevPaths) map[string]map[string]string {
+	msmss := map[string]map[string]string{}
+
+	for _, dp := range drivePrevs {
+		msmss[dp.id] = dp.folderIDToPrevPath
+	}
+
+	return msmss
+}
+
+// transforms 0 or more drivePrevPaths to a []data.RestoreCollection containing
+// a metadata collection.
+// DeltaURLs are currently always populated with {driveID: deltaURL()}.
+func multiDriveMetadata(
+	t *testing.T,
+	drivePrevs ...*drivePrevPaths,
+) []data.RestoreCollection {
+	drc := []data.RestoreCollection{}
```

Comment:
nit: could you please use a more descriptive name for this and `dp` below too too? This function is starting to get complex enough that meaningful names would be useful to the reader. Something like `prevMetadataCols` and `prevPaths` could work

Reply:
can do